snap/local/hooks/cmd/configure/configure.go

```diff
@@ -86,146 +109,262 @@ func serviceType(name string) int {
 //   - if env configuration for the service exists, use it to write a
 //     service-specific .env file to the service config dir in $SNAP_DATA
-// - start/stop any tightly couple services (e.g. if the secret-store
+// - start/stop any tightly coupled services (e.g. if the secret-store
 //   is disabled, the proxy also has to come down) if required
 // - start/stop the service itself if required
 //
 // NOTE - at this time, this function does *not* restart a service based
 // on env configuration changes. If changes are made after a service has
 // been started, the service must be restarted manually.
 //
-func handleAllServices() error {
+func handleAllServices(install bool) (error, []string) {
+	var startServices []string
+	var serviceList []string
+	secretStoreActive := true
 
 	// grab and log the current service configuration
 	for _, s := range hooks.Services {
 		var envJSON string
 
 		status, err := cli.Config(s)
 		if err != nil {
-			return err
+			return err, startServices
 		}
 
-		hooks.Debug(fmt.Sprintf("edgexfoundry:configure: service: %s status: %s", s, status))
+		hooks.Info(fmt.Sprintf("edgexfoundry:configure: %s service: %s status: %s", s, status))
 
 		serviceCfg := hooks.EnvConfig + "." + s
 		envJSON, err = cli.Config(serviceCfg)
 		if err != nil {
-			return fmt.Errorf("edgexfoundry:configure failed to read service %s configuration - %v", s, err)
+			err = fmt.Errorf("edgexfoundry:configure failed to read service %s configuration - %v", s, err)
+			return err, startServices
 		}
 
 		if envJSON != "" {
 			hooks.Debug(fmt.Sprintf("edgexfoundry:configure: service: %s envJSON: %s", s, envJSON))
 			if err := hooks.HandleEdgeXConfig(s, envJSON, nil); err != nil {
-				return err
+				return err, startServices
 			}
 		}
 
+		// SecBootstrapper is a valid service for configuration
+		// purposes, however it isn't individually controlable
+		// using on|off, so once configuration has been handled
+		// skip to the next service.
+		if s == hooks.ServiceSecBootstrapper {
+			continue
+		}
+
 		sType := serviceType(s)
 
 		switch sType {
-		case devVirtService:
-			hooks.Debug("edgexfoundry:configure: device-virtual")
-			// device-virtual is built with device-sdk-go which waits
-			// for core-data and core-metadata to come online, so if we are
-			// enabling a device service, we should also enable those services
-			if status == ON {
-				if err = handleSingleService("core-data", ON); err != nil {
-					return err
+		case kuiperService:
+			hooks.Debug("edgexfoundry:configure: kuiper")
+
+			switch status {
+			case INSTALL:
+				// FIXME: make this cli.UnsetConfig
+				if err := cli.SetConfig("kuiper", ""); err != nil {
+					return err, startServices
 				}
-				if err = handleSingleService("core-metadata", ON); err != nil {
-					return err
+				fallthrough
+			case ON:
+				serviceList = getKuiperServices()
+				if install {
+					startServices = appendServices(startServices, serviceList)
```

Comment:
this helper can go away completely if you do

```suggestion
					startServices = append(startServices, serviceList...)
```

Reply:
I guess even better you could do

```suggestion
					startServices = append(startServices, getKuiperServices()...)
```

Reply:
ah nevermind I see you do also need serviceList too, so ignore the second recommendation